---
title: "Intro to Transcriptomics"
author: Nicole Black, Wade Boohar, Kayla Xu
date: 07/17/22
updated: 09/26/23
---

***Deliverables***
-Upload this R Notebook to your GitHub and submit the link to your Repo on BB.
-Include any graphs or figures created in this assignment in the folder with your R notebook with descriptive file names.

Since this is an optional partner activity, it is okay if your answers are the same as your partnerâ€™s as long as everyone understands it and could explain it in their own words if asked. Each person must individually push their code to Github. *At the top of your R Notebook, write the name of you and your partner(s) as a comment.*

***Complete the following coding activity and answer any following questions as comments in your R Notebook***

In Transcriptomics_Tutorial.Rmd, you learned how to:

1. manipulate the SummarizedExperiment data structure, 
2. turn it into more readable dataframes, 
3. complete basic analysis using those dataframes, 
4. differential expression analysis based on age. 

Before saving the rna_counts, rna_clinical, and rna_genes however, we masked out any sample that was not a tumor.

In this assignment, you will need to reinstantiate the rna_se object and then perform differential expression analysis based on tumor status.

*Pre-Assignment*
Use knitr function to set your working directory to your analysis_data folder in qbio_490_name.
```{r}
#Set pathway
knitr::opts_knit$set(root.dir = normalizePath("/Users/tomasmanea/desktop/QBIO490/qbio_490_tomas/analysis_data")) 
```

Load all necessary packages and instantiate the rna_se object
```{r}
# install, only if needed, and load packages (SummarizedExperiment and TCGAbiolinks)
library(SummarizedExperiment)
library(TCGAbiolinks)
library(DESeq2)

rna_query <- GDCquery(project ="TCGA-BRCA",
                      data.category = "Transcriptome Profiling",
                      data.type = "Gene Expression Quantification",
                      workflow.type = "STAR - Counts")

rna_se <- GDCprepare(rna_query)

```


*1*
Perform all data cleaning and label manipulation steps (hint: applying colnames and rownames to make the dataframes more readable) except for masking out the patient with an NA for age and the "Solid Tissue Normal" sample types.
```{r}
##Masking for "rna_clinical, rna_counts, rna_genes 

rna_clinical <- rna_se@colData
rna_clinical <- as.data.frame(rna_clinical)

rna_counts <- rna_se@assays@data$unstranded
rna_counts <- as.data.frame(rna_counts)

rna_genes <- rna_se@rowRanges@elementMetadata
rna_genes <- as.data.frame(rna_genes)

##Subset "treatments", "primary site", and "disease_type"
treatments_mask <- ifelse(colnames(rna_clinical) == 'treatments', F, T)
rna_clinical <- rna_clinical[, treatments_mask]

primary_mask <- ifelse(colnames(rna_clinical) == 'primary_site', F, T)
rna_clinical <- rna_clinical[, primary_mask]

disease_mask <- ifelse(colnames(rna_clinical) == 'disease_type', F, T)
rna_clinical <- rna_clinical[, disease_mask]

#Writing csv file
write.csv(rna_clinical, "/Users/tomasmanea/desktop/QBIO490/qbio_490_tomas/analysis_data/brca_rna_clincial_data.csv", row.names = TRUE)

write.csv(rna_genes, "/Users/tomasmanea/desktop/QBIO490/qbio_490_tomas/analysis_data/brca_rna_gene_data.csv", row.names = TRUE)

write.csv(rna_counts, "/Users/tomasmanea/desktop/QBIO490/qbio_490_tomas/analysis_data/brca_rna_count_data.csv", row.names = TRUE)

# remove Solid Tissue Normal samples
rna_clinical_definition <- ifelse(rna_clinical$definition == "Solid Tissue Normal", F, T)
rna_clinical <- rna_clinical[rna_clinical_definition,]
rna_counts <- rna_counts[, rna_clinical_definition]
```


*2*
Determine and define your experimental design for the differential expression analysis. 
Control for covariance of age, gender, OR vital_status. 
Manipulate those columns so that they are ready for differential expression analysis 
(hint: what kind of variables are they? 
what data type are they by default? 
do you need to handle unknown values?)
```{r}
#Rownames & Colnames
rownames(rna_genes) <- rna_genes$gene_id
rownames(rna_counts) <- rna_genes$gene_id
colnames(rna_counts) <- rownames(rna_clinical)

#Mask creation - should get rid of values lower than 10 & get rid of any NAs in both covarianta and condition
vital_mask <- !is.na(rna_clinical$vital_status)
rna_clinical <- rna_clinical[vital_mask,]
rna_counts <- rna_counts[, vital_mask]
                       
gender_mask <- !is.na(rna_clinical$gender)
rna_clinical <- rna_clinical[gender_mask,]
rna_counts <- rna_counts[, gender_mask]
                        
count_mask <- ifelse(rowSums(rna_counts) < 10, F, T)
rna_genes <- rna_clinical[count_mask,]
rna_counts <- rna_counts[count_mask,]

#Experimental design - observing "tissue or organ of origin"; covariate being gender and vital status;

#rna_clinical into Factor & is.na (data type - char; Categorical variable)
rna_clinical$vital_status <- factor(rna_clinical$vital_status)
rna_clinical$gender <- factor(rna_clinical$gender)

#Running DESeq2
#DESeqDataSetFromMatrix
dds <- DESeqDataSetFromMatrix(countData = rna_counts,
                             colData = rna_clinical,
                             design = ~gender + vital_status )

#DESeq
dds_obj <- DESeq(dds) # note: this will likely take a long time (ie 2 hours - 3)

#resultsNames
resultsNames(dds_obj)  # see what comparisons got run

#Get the alive vs. dead comparison
results <- results(dds_obj, format = "DataFrame", contrast = c("vital_status", "Alive", "Dead")) 


```


*3*
Perform the differential expression analysis, create a volcano plot, and interpret what genes from each of the 6 sections means (assuming thresholds of -log10padj = 2 and log2FoldChange = 1)
```{r}
#Installing EnhancedVolcano
library(ggplot2)
library(EnhancedVolcano)


#Creating VolcanoPlot

EnhancedVolcano(res,
    lab = rownames(res),
    x = 'log2FoldChange',
    y = 'pvalue',
    title = 'N061011 versus N61311',
    pCutoff = 0.01,
    FCcutoff = 1,
    pointSize = 3.0,
    labSize = 6.0)

```

top-right genes: 
bottom-right genes: 
top-left genes: 
bottom-left genes: 
top-middle genes:
bottom-middle genes:

Save the picture of the volcano plot (using either ggsave() or right clicking and manually downloading the image and push this .Rmd and the image to GitHub)
